### PR TITLE
Improvement - Refactor Flotilla missions

### LINKED ISSUE #335: Merge Report & Scheduled Mission
**Describe the improvement you would like to see**
Flotilla currently has a troubled relationship to missions.
A Report is a FlotillaMission and a ScheduledMission is an EchoMission with a start time
A mission from the mission endpoint is an EchoMission and an EchoMission is really just a Mission Template.

The improvement is to simplify all this from the point of view of Flotilla.
The report and Scheduled mission are merged to a single "Mission" object that is the Flotilla Mission.
Echo missions are referred to as Mission templates within the backend.

**How will this change existing functionality?**
This will make our database more understandable and our API more intuitive to use.

**How will this improvement affect the current Threat Model?**
Will not affect threat model

### PR DESCRIPTION
**Pull request by GodVenn**
Closes #335

### CODE REVIEW

**Review by MortFred — COMMENTED** [review]
LGTM, but let's have another view on this from someone else in the team.
Large PR with lots of changes.

File: backend/api/Controllers/Models/CreateReportQuery.cs [hunk 1]
```diff
@@ -9,6 +9,6 @@ public struct CreateReportQuery
         public string EchoMissionId { get; set; }
         public string Log { get; set; }
         public DateTimeOffset StartTime { get; set; }
-        public ReportStatus ReportStatus { get; set; }
+        public MissionStatus ReportStatus { get; set; }
```
Comment (aestene) [inline backend/api/Controllers/Models/CreateReportQuery.cs:12]:
```suggestion
        public MissionStatus MissionStatus { get; set; }
```

File: backend/api/Database/Models/IsarTask.cs [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
         [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
-        [Required]
         public string Id { get; set; }
 
         [Required]
         public string IsarTaskId { get; set; }
 
         [Required]
-        public virtual Report Report { get; set; }
+        public virtual Mission Report { get; set; }
```
Comment (aestene) [inline backend/api/Database/Models/IsarTask.cs:19]:
```suggestion
        public virtual Mission Mission { get; set; }
```

**Review by aestene — APPROVED** [review]
LGTM

**Pull request merged**